### PR TITLE
[JENKINS-29340] ISVNAuthentication provider did not provide credentials

### PR DESCRIPTION
**Pull request by recena**
https://issues.jenkins-ci.org/browse/JENKINS-29340

### CODE REVIEW

**Comment by recena** [comment]
@reviewbybees Could you review these changes? /cc @christ66

**Comment by recena** [comment]
In my local development environment are working fine.

**Comment by christ66** [comment]
:bee: . I executed this pull request on my local machine and was unable to reproduce the failure.

File: src/main/java/hudson/scm/SubversionSCM.java [hunk 1]
```diff
@@ -1392,7 +1392,13 @@ else if (project.getLastBuild()!=null) {
 
         final Map<String,ISVNAuthenticationProvider> authProviders = new LinkedHashMap<String,
                 ISVNAuthenticationProvider>();
-        for (ModuleLocation loc: getLocations()) {
+
+        EnvVars env = null;
+        if (project.getLastCompletedBuild() != null) {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/scm/SubversionSCM.java]: Wrong construct, which will make the static analysis angry (e.g. cleanup of builds between two calls). You should assign the last completed build to a variable and then check&use it

Comment (recena) [inline src/main/java/hudson/scm/SubversionSCM.java]: @oleg-nenashev Before to do any modifications, I need to understand what do you mean. What static analysis tool?

Comment (oleg-nenashev) [inline src/main/java/hudson/scm/SubversionSCM.java]: Any.
Result of "project.getLastCompletedBuild()" may change between two calls => such null check does not guarantee the result

**Comment by recena** [comment]
@oleg-nenashev Thanks for your comment. Reviewed.

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by daniel-beck** [comment]
This change does not consider environment variables changed since the previous build. As replacement to #83 it is unsuitable. (And FWIW I consider reuse of previous builds' environments to be an antipattern.)

**Comment by recena** [comment]
@daniel-beck I agree. But I didn't know how to get the environment vars (all of them). Any idea?

**Comment by recena** [comment]
@daniel-beck I'm going to work in order to improve the current merged PR.

**Pull request merged**